picklerick/src/App.js

```diff
@@ -0,0 +1,224 @@
+import React from 'react';
+import './App.css';
+import Characters from './Characters';
+import { BrowserRouter as Router, Route,Switch} from 'react-router-dom';
+import Logo from './Rick_and_Morty_-_logo_.png';
+import CharacterDetails from "./CharacterDetails";
+import EpisodeDetails from './EpisodeDetails';
+import ButtonGroup from "react-bootstrap/es/ButtonGroup";
+import {  Link} from 'react-router-dom';
+const FA = require('react-fontawesome');
+
+
+/**
+ * Respond to clear event in the search box
+ */
+class SearchBox extends React.Component {
+    constructor(props) {
+        super(props);
+        this.input = React.createRef();
+    }
+
+    componentDidMount() {
+        this.input.current.addEventListener('search', this.onSearch);
+    }
+
+    onSearch = (e) => {
+
+        window.location.href = "/";
+
+    }
+
+    render() {
+        return (
+            <div>
+                <input className="form-control mr-sm-2" type="search" placeholder="Search by name" id="txtSearchString" aria-label="Search" ref={this.input} />
+            </div>
+        )
+    }
+}
+
+class App extends  React.Component{
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+
+            info: {},
+            data: {},
+            isLoading: true,
+            hasError: false,
+            total:0
+        }
+    }
+    componentDidMount(){
+        fetch('https://rickandmortyapi.com/api/character')
+            .then(response => {
+                return response.json();
+            })
+            .then(data => {
+                this.setState({
+                    info: data.info,
+                    data: data.results,
+                    isLoading:false
+                })
+            })
+            .catch(error=>
+            {
+                this.setState({
+                    hasError:true,
+                    isLoading:false
+                });
+            })
+    }
+
+    fetchNextResultSet = (e) => {
+        const searchString = document.getElementById('txtSearchString');
+
+        let url;
+        if (e.currentTarget.id === 'next'){
+            if (this.state.info.next !== ""){
+                url=this.state.info.next;
+            }
+        }
+        else if(e.currentTarget.id === 'prev'){
+            if (this.state.info.prev !== ""){
+                url=this.state.info.prev;
+            }
+        }
+        else if(e.currentTarget.id === 'last'){
+            url=`https://rickandmortyapi.com/api/character/?page=${this.state.info.pages}`;
+        }
+        else if(e.currentTarget.id === 'first'){
+            url="https://rickandmortyapi.com/api/character/?page=1";
+        }
+        else if(e.currentTarget.id ==='btnSearch'){
+            if (searchString.value !==""){
```

Comment:
Nice workaround. It looks like hitting enter on an empty search still refreshes the page though?

Reply:
Maybe add an if searchString.value === "" then return so the fetch and display never occurs.